src/library_html5.js

```diff
@@ -3269,4 +3237,56 @@ var LibraryJSEvents = {
   }
 };
 
+function handleWebGLProxying(funcs) {
+  if (!USE_PTHREADS) return; // No proxying needed in singlethreaded builds
+
+  function listOfNFunctionArgs(func) {
+    var args = [];
+    for(var i = 0; i < func.length; ++i) {
+      args.push('p' + i);
+    }
+    return args;
+  }
+
+  for(var i in funcs) {
+    if (funcs[i + '__proxy'] == 'sync_on_webgl_context_handle_thread') {
+#if OFFSCREENCANVAS_SUPPORT
+      // With OffscreenCanvas builds, GL context handle may be owned by main thread, the calling pthread,
+      // or another pthread. Route the call to the right thread.
+      // TODO: this handles the calling pthread and main thread cases, but not yet the case from pthread->pthread.
+      funcs[i + '_calling_thread'] = funcs[i];
+      funcs[i + '_main_thread'] = i + '_calling_thread';
+      funcs[i + '_main_thread__proxy'] = 'sync';
+      funcs[i + '_main_thread__sig'] = funcs[i + '__sig'];
+      if (!funcs[i + '__deps']) funcs[i + '__deps'] = [];
+      funcs[i + '__deps'].push(i + '_calling_thread');
+      funcs[i + '__deps'].push(i + '_main_thread');
+      delete funcs[i + '__proxy'];
+      var funcArgs = listOfNFunctionArgs(funcs[i]);
+      var funcArgsString = funcArgs.join(',');
+      var funcBody = `return GL.contexts[p0] ? _${i}_calling_thread(${funcArgsString}) : _${i}_main_thread(${funcArgsString});`;
+      if (funcs[i + '_before_on_calling_thread']) {
+        funcs[i + '__deps'].push(i + '_before_on_calling_thread');
+        funcBody = `_${i}_before_on_calling_thread(${funcArgsString}); ` + funcBody;
+      }
+      funcArgs.push(funcBody);
+      funcs[i] = new (Function.prototype.bind.apply(Function, [Function].concat(funcArgs)));
+#else
+#if OFFSCREEN_FRAMEBUFFER
+      // When building with OFFSCREEN_FRAMEBUFFER but without OFFSCREENCANVAS_SUPPORT,
+      // only main thread creates WebGL contexts, so all calls are unconditionally proxied.
+      funcs[i + '__proxy'] = 'sync';
+#else
+      // Building without OFFSCREENCANVAS_SUPPORT or OFFSCREEN_FRAMEBUFFER: the application
+      // will only utilize WebGL in the main browser thread. Remove any WebGL proxying
```

Comment:
this says "any" but it only removes the special new `sync_on_webgl_context_handle_thread` proxying?